### DISCUSSION TITLE
any ways to link tagging w/ tree structure (filing) - what would the community say?

### DISCUSSION THREAD

**Discussion by lerone-b** [TriliumNext/Trilium#9405] [Q&A]
an idea for linking tag-system with the hierarchical filing system:
it would be great **if _cloning_ a note to another note could effectively apply a (searchable) tag/attribute** to a note – ... whatever the way. 
– so I am wondering whether there is any way to achieve that in the given architecture of Trilium?

there were **already some observations/inquiries about searching via _'inherited attributes'_ and** also mentioning of **_'copy inheritance'_ in that larger context** (see #606 + #760 + #850 ) – both of which seem to be possible enablers/pivots for something like what I am thinking about here. 

– now: '_inherited attributes_' are not searchable in Trilium it seems; and with the '_copy inheritance_' it seems after testing here that this does not work in this scenario either; 'copy inheritance' only seems to take effect on notes _originally created as childs_, not those placed later as clones (- right?)

... in any way + to give the reasoning behind that: 
**some way of tagging directly linkable to the tree structure** (effectively linking tagging and hierarchical 'filing') would be **terrifically useful** as it would e.g.
- allow for flexible, personally 'customizable' representation of the tag structure in the tree as that can be re-sorted, cloned, folded in whatever way (thus indirectly **allowing things like tag-hierarchies, different sets of controlled vocabularies etc.**)
- leverage note **cloning as a quick form of tagging**
...

wanted to raise this from another angle, especially after I have seen some other similarly motivated, but unresolved threads here (see above)...

(PS – don´t know whether this is obsolete with the implementation of attribute search as part of the global search ( #980 ), but I guess not...?)

**Comment by lerone-b** [comment]
... another way at this ( – think: '**asking the community**'):

searching around for ways to get somewhere close to my idea of linking tagging (i.e. attributes) with the tree structure (and re-filing/cloning a note), it seems one way would go via **scripting** and the API. 
the model of the task note scripts that file tasks for example once their location is changed seems pretty much the same thing in another context (and it seems mainly dependent on **'runOnAttributeChange'**).

as I am _no coder_ and do not really know where to start with fiddling in javascript, I wonder if anyone has dealt with this before? and can share experience and/or hints **how to achieve some specific re-filing (cloning) of a note on feeding the tag-attribute with a certain value**.

personally, I couldn´t extract the right way to do it out of the related code-note within the demo-content – '../task manager/implementation/attribute changed'.
... so, I guess what I am trying here is seeing whether I can tap somehow into the community knowledge; or activate something like the mentioned in #986 (Community Code Library)...?! :-)

I´d even be happy to be pointed at **some more elaborate documentation** that could help me make my first steps in java-coding here and work with scripting in Trilium and/or the API...

**appreciate any hand, help or hint on this! thx in advance!**
oliver

**Comment by zadam** [comment]
I guess the searchable inherited attributes would be the best approach for this and it would be certainly an attractive feature. The problem is that it is quite technically challenging ...

**Comment by lerone-b** [comment]
thx Zadam,

yes, I copied that as a problem from one of the referenced threads. and of course simply accept your word/call/judgement on this.
I was under the assumption that **a customized/modified adaption of the current behavior of your task-examples** – in which tasks are indeed 're-filed' in the tree after selection of a location attribute – would be an alternative route. but then I might have limited grasp...
I also figured this works mainly via 'runOnAttributeChange' + scripting. as this would **circumvent**, in my limited understanding, **the need for searchable inherited attributes** I thought I ask people who know...

– is my thinking on this 'alternative' way misguided then?
... I am following up on that because I think if such a thing can be achieved it immediately opens up **a whole new set of options to productively work with tags, w/o any need of touching/modifying the tag-system itself**...
